File: react/src/api/farms-api.ts
import type { Classification } from '../types/classification';
import type { Farm } from '../types/farm';
import { fetchApi } from './fetch-api';

export const getFarms = (
  country?: string,
  county?: string,
  classification?: Classification,
  operating?: boolean
) =>
  fetchApi<Farm[]>('GET', '/api/farms', {
    country,
    county,
    classification,
    operating,
  });

File: flutter/lib/api/farms_api.dart
import 'package:livetec_flutter_app/api/fetch_api.dart';
import 'package:livetec_flutter_app/types/classification.dart';
import 'package:livetec_flutter_app/types/farm.dart';
import 'package:livetec_flutter_app/utils/string.dart';

Future<List<Farm>> getFarms(
  String? country,
  String? county,
  Classification? classification,
  bool? operating,
) async {
  final data = await dioService.fetch(
    FetchMethod.get,
    '/api/farms',
    query: {
      if (country != null) 'country': capitalizeFirstLetter(country),
      if (county != null) 'county': capitalizeFirstLetter(county),
      if (classification != null) 'classification': classification.value,
      if (operating != null) 'operating': operating,
    },
  );

  if (data == null) return [];
  return (data as List).map((element) => Farm.fromJson(element)).toList();
}
